code/game/machinery/constructable_frame.dm

```diff
@@ -111,7 +112,8 @@
 					icon_state = "box_2"
 					state = 3
 					components = list()
-					req_components = circuit.req_components.Copy()
+					if(circuit.req_components)
+						req_components = circuit.req_components.Copy()
```

Comment:
```suggestion
					req_components = circuit.req_components?.Copy()
```

Reply:
It should actually be `req_components = circuit?.req_components.Copy()`
because you want to check the `.req_components` not the `.Copy()`